test/common/grpc/async_client_manager_impl_test.cc

```diff
@@ -17,14 +18,17 @@ namespace {
 
 class AsyncClientManagerImplTest : public testing::Test {
 public:
+  AsyncClientManagerImplTest() : api_(std::chrono::milliseconds(1000)) {}
+
   Upstream::MockClusterManager cm_;
   NiceMock<ThreadLocal::MockInstance> tls_;
   Stats::MockStore scope_;
   DangerousDeprecatedTestTime test_time_;
+  Api::Impl api_;
```

Comment:
Should this be a `MockApi`?

Reply:
I don't think so. The AsyncClientManagerImpl allocates a GoogleAsyncClientThreadLocal object which creates a thread and then its destructor joins on that thread. The MockApi currently returns null when calling createThread so that causes a crash. I suppose I could add a MockThread object and have MockApi::createThread return that. Is it worth it?